module/core/src/Application/Request/ParamConverter/AggregateRootParamConverter.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/**
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+declare(strict_types = 1);
+
+namespace Ergonode\Core\Application\Request\ParamConverter;
+
+use Ergonode\EventSourcing\Domain\AbstractAggregateRoot;
+use Ergonode\EventSourcing\Infrastructure\Manager\EventStoreManager;
+use Ergonode\SharedKernel\Domain\AggregateId;
+use Ergonode\Workflow\Domain\Entity\AbstractWorkflow;
+use Ergonode\Workflow\Domain\Provider\WorkflowProvider;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\ParamConverter;
+use Sensio\Bundle\FrameworkExtraBundle\Request\ParamConverter\ParamConverterInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Webmozart\Assert\Assert;
+
+/**
+ */
+class AggregateRootParamConverter implements ParamConverterInterface
+{
+    /**
+     * @var EventStoreManager
+     */
+    private EventStoreManager $manager;
+
+    /**
+     * @param EventStoreManager $manager
+     */
+    public function __construct(EventStoreManager $manager)
+    {
+        $this->manager = $manager;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function apply(Request $request, ParamConverter $configuration): void
+    {
+        $name = $configuration->getName();
+        Assert::notNull($name);
+
+        $parameter = $request->get($name);
+        Assert::notNull($parameter);
+
+        if (!AggregateId::isValid($parameter)) {
+            throw new BadRequestHttpException(sprintf('Invalid "%s" ID', $name));
+        }
+        $resource = $this->manager->load(new AggregateId($parameter));
+
+        if (null === $resource) {
+            throw new NotFoundHttpException(sprintf('"%s" by ID "%s" not found', $name, $parameter));
+        }
+        $class = $configuration->getClass();
+
+        if (!$resource instanceof $class) {
+            throw new NotFoundHttpException(sprintf('Resource is not instance of %s', $name));
```

Comment:
Info should be sth more like that "there's no such user resource" or just "no such resource found"